### ISSUE TITLE
is it a bug in calculating per_token loss?

### ISSUE THREAD

**Issue by weizhenli012** [om-ai-lab/VLM-R1#38]
# x - x.detach() allows for preserving gradients from x
per_token_loss = torch.exp(per_token_logps - per_token_logps.detach()) * advantages.unsqueeze(1)

torch.exp(per_token_logps - per_token_logps.detach()) both are per_token_logps, the second should be derived from the old policy(model)?

**Comment by xrc10** [comment]
This is a simplification of the GRPO in the original paper, by assuming 𝜇=1. In that case the old policy model is the same as the current one.
